lib/web_ui/dev/watcher.dart

```diff
@@ -221,31 +225,35 @@ class PipelineWatcher {
     });
   }
 
-  void _runPipeline() {
-    int runCount;
-    switch (pipeline.status) {
-      case PipelineStatus.started:
-        pipeline.stop().then((_) {
-          runCount = _pipelineRunCount;
-          pipeline.start().then((_) => _pipelineDone(runCount));
-        });
-        break;
-
-      case PipelineStatus.stopping:
-        // We are already trying to stop the pipeline. No need to do anything.
-        break;
-
-      default:
-        runCount = _pipelineRunCount;
-        pipeline.start().then((_) => _pipelineDone(runCount));
-        break;
+  void _runPipeline() async {
+    if (pipeline.status == PipelineStatus.stopping) {
+      // We are already trying to stop the pipeline. No need to do anything.
+      return;
+    }
+
+    if (pipeline.status == PipelineStatus.started) {
+      // If the pipeline already running, stop it before starting it again.
+      await pipeline.stop();
+    }
+
+    final int runCount = _pipelineRunCount;
+    try {
+      await pipeline.run();
+      _pipelineSucceeded(runCount);
+    } catch(error, stackTrace) {
+      _pipelineFailed(error, stackTrace);
```

Comment:
Can you add comment explaining why a rethrow isn't needed here?

Reply:
Done.